### PR TITLE
Icon Component

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-450

### CODE REVIEW

**Review by creed-victor — APPROVED** [review]

File: packages/ui/src/1_atoms/Icon/Icon.tsx [hunk 1]
```diff
@@ -76,6 +76,7 @@ export const Icon: React.FC<IconProps> = ({
         width={iconSize}
         fill="currentColor"
         className={classNames(className, inlineBlock)}
+        data-icon={icon}
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/Icon.tsx:85]: I know it's not specifically mentioned in the description but we had a discussion with @soulBit and we want to have `data-action-id` in all of the components so pls add it to this component as well (take a look at `Input` and `InputBase` in https://github.com/DistributedCollective/sovryn-dapp/pull/8 ).

File: packages/ui/src/1_atoms/Icon/icon.test.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      fillRule="evenodd"
+    ></path>
+  </svg>
+);
+
+describe('Test for all icon types', () => {
+  //render icon from the Fontawesome library
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/icon.test.jsx]: We don't need these kinds of comments, the test names should be descriptive enough. Pls delete them.

File: packages/ui/src/1_atoms/Icon/icon.test.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  test('render fa icon', () => {
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/icon.test.jsx]:
```suggestion
  test('renders an fa icon', () => {
```

File: packages/ui/src/1_atoms/Icon/icon.test.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    render(<Icon icon={faBacterium} size="2x" />);
+    expect(screen.findAllByRole('svg[data-icon="bacterium"]')).toBeDefined();
+  });
+
+  //render custom icon
+  test('render custom icon', () => {
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/icon.test.jsx]:
```suggestion
  test('renders a custom icon', () => {
```

File: packages/ui/src/1_atoms/Icon/icon.test.jsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    render(<Icon icon={customIcon} />);
+    expect(screen.findAllByRole('svg[data-icon="customIcon"]')).toBeDefined();
+  });
+
+  //render icon from our Sovryn Library
+  test('render icon from Sovryn Library', () => {
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/icon.test.jsx]:
```suggestion
  test('renders an icon from Sovryn Library', () => {
```

**Review by tiltom — CHANGES_REQUESTED** [review]

File: packages/ui/src/1_atoms/Icon/Icon.tsx [hunk 6]
```diff
@@ -77,6 +83,7 @@ export const Icon: React.FC<IconProps> = ({
         fill="currentColor"
         className={classNames(className, inlineBlock)}
         data-icon={icon}
+        data-action-id={dataActionId}
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/Icon.tsx:86]: You have a missing dependency in `renderIcon`.

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — APPROVED** [review]

**Pull request merged**